src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java

```diff
@@ -0,0 +1,134 @@
+import net.snowflake.ingest.utils.Constants;
+
+/**
+ * Object to convert and store properties for {@link
+ * net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient}. This object is used to compare
+ * equality between clients in {@link StreamingClientProvider}.
+ */
+public class StreamingClientProperties {
+  public static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT_";
+  public static final String DEFAULT_CLIENT_NAME = "DEFAULT_CLIENT";
+
+  private static final KCLogger LOGGER = new KCLogger(StreamingClientProperties.class.getName());
+
+  // contains converted config properties that are loggable (not PII data)
+  public static final List<String> LOGGABLE_STREAMING_CONFIG_PROPERTIES =
+      Stream.of(
+              Constants.ACCOUNT_URL,
+              Constants.ROLE,
+              Constants.USER,
+              StreamingUtils.STREAMING_CONSTANT_AUTHORIZATION_TYPE)
+          .collect(Collectors.toList());
+
+  public final Properties clientProperties;
+  public final String clientName;
+  public final Map<String, Object> parameterOverrides;
+
+  /**
+   * Creates non-null properties, client name and parameter overrides for the {@link
+   * net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient} from the given connectorConfig
+   * Properties are created by {@link StreamingUtils#convertConfigForStreamingClient(Map)} and are a
+   * subset of the given connector configuration
+   *
+   * @param connectorConfig Given connector configuration. Null configs are treated as an empty map
+   */
+  public StreamingClientProperties(Map<String, String> connectorConfig) {
+    // treat null config as empty config
+    if (connectorConfig == null) {
+      LOGGER.warn(
+          "Creating empty streaming client properties because given connector config was empty");
+      connectorConfig = new HashMap<>();
+    }
+
+    this.clientProperties = StreamingUtils.convertConfigForStreamingClient(connectorConfig);
+
+    this.clientName =
+        STREAMING_CLIENT_PREFIX_NAME
+            + connectorConfig.getOrDefault(Utils.NAME, DEFAULT_CLIENT_NAME);
+
+    // Override only if bdec version is explicitly set in config, default to the version set
+    // inside Ingest SDK
+    this.parameterOverrides = new HashMap<>();
+    Optional<String> snowpipeStreamingBdecVersion =
+        Optional.ofNullable(connectorConfig.get(SNOWPIPE_STREAMING_FILE_VERSION));
+    snowpipeStreamingBdecVersion.ifPresent(
+        overriddenValue -> {
+          LOGGER.info("Config is overridden for {} ", SNOWPIPE_STREAMING_FILE_VERSION);
+          parameterOverrides.put(BLOB_FORMAT_VERSION, overriddenValue);
+        });
+  }
+
+  /**
+   * Gets the loggable properties, see {@link
+   * StreamingClientProperties#LOGGABLE_STREAMING_CONFIG_PROPERTIES}
+   *
+   * @return A formatted string with the loggable properties
+   */
+  public String getLoggableClientProperties() {
+    return this.clientProperties == null | this.clientProperties.isEmpty()
```

Comment:
can `clientProperties` be null or empty? I thought we have check to make sure some of the configurations are required?

Reply:
it shouldn't ever be null or empty, but I added the check just in case